### PR TITLE
CP-13254: SR-level RRDs: Add API and CLI calls to query SR-level RRDs

### PR DESCRIPTION
**Pull request by koushikcgit**
Adding APIs to query the SR Level RRD stats

Signed-off-by: Koushik Chakravarty <koushik.chakravarty@citrix.com>

### CODE REVIEW

**Comment by xen-git** [comment]
Can one of the admins verify this patch for testing?

**Comment by johnelse** [comment]
@xen-git test this please

**Comment by sharady** [comment]
@xen-git retest this please

**Comment by johnelse** [comment]
@sharady please could you rebase this against the latest master branch? That should allow the travis build to pass.

File: ocaml/xapi/message_forwarding.ml [hunk 1]
```diff
@@ -3197,6 +3197,23 @@ module Forward = functor(Local: Custom_actions.CUSTOM_ACTIONS) -> struct
...
+			Local.SR.record_data_source ~__context ~sr ~data_source
+
+		let query_data_source ~__context ~sr ~data_source =
+			info "SR.query_data_source: SR = '%s'; data source = '%s'" (sr_uuid ~__context sr) data_source;
+			Local.SR.query_data_source ~__context ~sr ~data_source
+
+		let forget_data_source_archives ~__context ~sr ~data_source =
+			info "SR.forget_data_source_archives: sr = '%s'; data source = '%s'" (sr_uuid ~__context  sr) data_source;
+			Local.SR.forget_data_source_archives ~__context ~sr ~data_source
+
```
Comment (johnelse) [inline ocaml/xapi/message_forwarding.ml]: These operations should all be forwarded with `forward_sr_op` - see the implementation of SR.update for an example.

**Comment by johnelse** [comment]
Other than the missing forwarding, this looks good.

**Comment by johnelse** [comment]
Superceded by https://github.com/xapi-project/xen-api/pull/2356

**Pull request closed without merging**